### ISSUE TITLE
[Req] Z3 theorem prover

### ISSUE THREAD

**Issue by ArsenShnurkov** [gentoo/dotnet#205]
Z3 is a theorem prover from Microsoft Research. It is licensed under the MIT license.
https://github.com/Z3Prover/z3

The default Z3 bindings for .NET are built for the .NET framework version 4
https://github.com/Z3Prover/z3/tree/master/src/api/dotnet

**Comment by mpkh** [comment]
@ArsenShnurkov are you going working on this?

**Comment by ArsenShnurkov** [comment]
no. Req - it's then i request [b]somebody[/b] other that it would be nice to have something.
